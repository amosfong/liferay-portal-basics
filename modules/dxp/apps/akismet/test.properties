##
## Test Suites
##

    modified.files.includes[relevant][akismet-functional-rule]=**

    modified.files.includes[relevant][akismet-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=\
        akismet-functional-rule,\
        akismet-java-rule

    test.batch.names[relevant][akismet-functional-rule]=\
        functional-tomcat90-mysql57

    test.batch.names[relevant][akismet-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]}

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]=\
        (testray.main.component.name ~ "Akismet")

##
## Testray
##

    testray.main.component.name=Akismet