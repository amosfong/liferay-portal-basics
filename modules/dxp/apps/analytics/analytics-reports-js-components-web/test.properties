##
## Test Suites
##

    modified.files.includes[relevant][analytics-reports-js-components-web-playwright-rule]=**
    relevant.rule.names=analytics-reports-js-components-web-playwright-rule

    test.batch.names[relevant][analytics-reports-js-components-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][analytics-reports-js-components-web-playwright-rule]=\
        analytics-reports-js-components-web