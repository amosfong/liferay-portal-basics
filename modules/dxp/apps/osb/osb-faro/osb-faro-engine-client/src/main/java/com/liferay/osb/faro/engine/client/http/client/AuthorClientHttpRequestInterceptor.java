/**
 * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.engine.client.http.client;

import com.liferay.osb.faro.engine.client.constants.OSBAsahHeaderConstants;
import com.liferay.portal.kernel.model.User;
import com.liferay.portal.kernel.security.permission.PermissionChecker;
import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;

import java.io.IOException;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpRequest;
import org.springframework.http.client.ClientHttpRequestExecution;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.http.client.ClientHttpResponse;

/**
 * @author Marcellus Tavares
 */
public class AuthorClientHttpRequestInterceptor
	implements ClientHttpRequestInterceptor {

	@Override
	public ClientHttpResponse intercept(
			HttpRequest httpRequest, byte[] bytes,
			ClientHttpRequestExecution clientHttpRequestExecution)
		throws IOException {

		_addAuthorHeaders(httpRequest);

		return clientHttpRequestExecution.execute(httpRequest, bytes);
	}

	private void _addAuthorHeaders(HttpRequest httpRequest) {
		User user = _getUser();

		if (user == null) {
			return;
		}

		HttpHeaders httpHeaders = httpRequest.getHeaders();

		httpHeaders.add(
			OSBAsahHeaderConstants.AUTHOR_USER_ID,
			String.valueOf(user.getUserId()));
		httpHeaders.add(
			OSBAsahHeaderConstants.AUTHOR_USER_NAME, user.getFullName());
	}

	private User _getUser() {
		PermissionChecker permissionChecker =
			PermissionThreadLocal.getPermissionChecker();

		if (permissionChecker == null) {
			return null;
		}

		return permissionChecker.getUser();
	}

}