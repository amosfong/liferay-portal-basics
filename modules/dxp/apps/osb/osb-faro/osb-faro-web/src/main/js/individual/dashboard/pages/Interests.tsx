import BasePage from 'shared/components/base-page';
import ClayLink from '@clayui/link';
import Interests from '../hocs/Interests';
import React from 'react';
import StatesRenderer from 'shared/components/states-renderer/StatesRenderer';
import URLConstants from 'shared/util/url-constants';
import {Routes, toRoute} from 'shared/util/router';
import {useCurrentUser} from 'shared/hooks/useCurrentUser';
import {useDataSource} from 'shared/hooks/useDataSource';
import {useParams} from 'react-router-dom';

const InterestsPage = () => {
	const {groupId} = useParams();
	const currentUser = useCurrentUser();
	const authorized = currentUser.isAdmin();

	const dataSourceStates = useDataSource();

	return (
		<BasePage.Body pageContainer>
			<StatesRenderer {...dataSourceStates}>
				<StatesRenderer.Empty
					description={
						<>
							{authorized
								? Liferay.Language.get(
										'connect-a-data-source-with-sites-data'
								  )
								: Liferay.Language.get(
										'please-contact-your-workspace-administrator-to-add-data-sources'
								  )}

							<ClayLink
								className='d-block mb-3'
								href={URLConstants.DataSourceConnection}
								key='DOCUMENTATION'
								target='_blank'
							>
								{Liferay.Language.get(
									'access-our-documentation-to-learn-more'
								)}
							</ClayLink>

							{authorized && (
								<ClayLink
									button
									className='button-root'
									displayType='primary'
									href={toRoute(
										Routes.SETTINGS_ADD_DATA_SOURCE,
										{
											groupId
										}
									)}
								>
									{Liferay.Language.get(
										'connect-data-source'
									)}
								</ClayLink>
							)}
						</>
					}
					displayCard
					title={Liferay.Language.get(
						'no-sites-synced-from-data-sources'
					)}
				/>

				<StatesRenderer.Success>
					<div className='individuals-dashboard-interests-root'>
						<Interests />
					</div>
				</StatesRenderer.Success>
			</StatesRenderer>
		</BasePage.Body>
	);
};

export default InterestsPage;
