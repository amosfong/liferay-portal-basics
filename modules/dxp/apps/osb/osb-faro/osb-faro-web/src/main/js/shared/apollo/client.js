import cache from './cache';
import {ApolloClient} from 'apollo-client';
import {DEVELOPER_MODE} from 'shared/util/constants';
import {get} from 'lodash';
import {HttpLink} from 'apollo-link-http';
import {onError} from 'apollo-link-error';
import {reloadPage} from 'shared/util/router';
import {resolvers} from './resolvers/resolvers';

const groupIdRegex = /^\/workspace\/([a-z0-9._-]+)/;

/**
 * Fecth With Project ID
 * @param {string} uri
 * @param {object} options
 */
const fetchWithGroupId = (uri, options) => {
	const currentUri = new URL(window.location.href);
	const {operationName} = JSON.parse(options.body);
	const matches = currentUri.pathname.match(groupIdRegex);

	if (matches !== null && matches.length > 1) {
		const groupId = matches[1];

		return fetch(
			`${uri}?opname=${operationName}&projectGroupId=${groupId}`,
			options
		);
	}

	return fetch(`${uri}/&opname=${operationName}`, options);
};

const client = new ApolloClient({
	addTypename: true,
	cache,
	connectToDevTools: FARO_DEV_MODE,
	defaultOptions: {
		watchQuery: {
			notifyOnNetworkStatusChange: true
		}
	},
	link: HttpLink.from([
		onError(({operation}) => {
			const status = get(operation.getContext(), ['response', 'status']);

			if (status === 401) {
				reloadPage();
			}
		}),
		new HttpLink({
			credentials: 'same-origin',
			fetch: fetchWithGroupId,
			uri: '/o/cerebro/graphql'
		})
	]),

	resolvers: {
		/**
		 * Queries must render only in the dev mode to avoid
		 * add unnecessary code in the final bundle
		 */
		Query: DEVELOPER_MODE ? resolvers : {}
	}
});

export default client;
