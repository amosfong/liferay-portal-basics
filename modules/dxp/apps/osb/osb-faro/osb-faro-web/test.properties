##
## Test Suites
##

    modified.files.includes[relevant][osb-faro-web-playwright-rule]=**
    relevant.rule.names=osb-faro-web-playwright-rule

    test.batch.names[relevant][osb-faro-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][osb-faro-web-playwright-rule]=\
        osb-faro-web

##
## Testray
##

    testray.main.component.name=Analytics Cloud