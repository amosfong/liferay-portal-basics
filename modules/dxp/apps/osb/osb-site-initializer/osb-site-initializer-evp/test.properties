##
## Test Suites
##

    modified.files.includes[relevant][osb-site-initializer-evp-functional-rule]=**
    relevant.rule.names=osb-site-initializer-evp-functional-rule

    test.batch.names[relevant][osb-site-initializer-evp-functional-rule]=\
        functional-tomcat90-mysql57

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP