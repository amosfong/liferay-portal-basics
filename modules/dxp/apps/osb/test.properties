##
## Test Suites
##

    modified.files.includes[relevant][osb-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=osb-java-rule

    test.batch.names[relevant][osb-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

##
## Testray
##

    testray.main.component.name=Analytics Cloud