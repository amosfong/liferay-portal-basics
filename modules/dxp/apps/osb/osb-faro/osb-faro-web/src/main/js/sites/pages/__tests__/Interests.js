import client from 'shared/apollo/client';
import Interests from '../Interests';
import React from 'react';
import {ApolloProvider} from '@apollo/react-components';
import {createMemoryHistory} from 'history';
import {MockedProvider} from '@apollo/react-testing';
import {mockPreferenceReq, mockTimeRangeReq} from 'test/graphql-data';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {waitForLoadingToBeRemoved} from 'test/helpers';

jest.unmock('react-dom');

const WrapperComponent = () => {
	const history = createMemoryHistory();

	return (
		<ApolloProvider client={client}>
			<Router history={history}>
				<MockedProvider
					mocks={[mockTimeRangeReq(), mockPreferenceReq()]}
				>
					<Interests router={{params: {}, query: {}}} />
				</MockedProvider>
			</Router>
		</ApolloProvider>
	);
};

describe('Sites Dashboard Interests', () => {
	it('render', async () => {
		const {container} = render(<WrapperComponent />);

		await waitForLoadingToBeRemoved(container);

		expect(container).toMatchSnapshot();
	});
});
