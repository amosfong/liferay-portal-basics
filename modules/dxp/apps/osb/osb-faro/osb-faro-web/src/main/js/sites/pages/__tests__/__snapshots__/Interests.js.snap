// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sites Dashboard Interests render 1`] = `
<div>
  <div
    class="sites-dashboard-interests-root"
  >
    <div
      class="row"
    >
      <div
        class="col-xl-12"
      >
        <div
          class="card card-root sites-interests-root page-display"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <div
              class="dropdown dropdown-range-key-root"
            >
              <button
                aria-controls="clay-dropdown-menu-1"
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle button-root btn btn-outline-borderless btn-sm btn-outline-secondary"
                type="button"
              >
                Last 30 days
                <svg
                  class="lexicon-icon lexicon-icon-caret-bottom icon-root ml-2"
                  role="presentation"
                >
                  <use
                    href="#caret-bottom"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="error-display-root flex-grow-1 error-spacer no-results-root flex-grow-1"
          >
            <div
              class="no-results-content"
            >
              <div
                class="h4 no-results-title"
              >
                An unexpected error occurred.
              </div>
              <button
                class="button-root btn btn-secondary"
                type="button"
              >
                Reload
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
